<?xml version="1.0"?>
<robot name="robotic_arm_inspector" xmlns:xacro="http://ros.org/wiki/xacro">
  <!-- ur5 from universal robots package-->
  <xacro:include filename="$(find ur_description)/urdf/common.gazebo.xacro" />
  <xacro:include filename="$(find ur_description)/urdf/ur5.urdf.xacro" />
  <!-- kinect -->
  <xacro:include filename="$(find common_sensors)/urdf/sensors/kinect.urdf.xacro" />
  <!-- <xacro:include filename="$(find common_sensors)/urdf/sensors/kinect2.urdf.xacro" /> -->

  <link name="world" />

  <joint name="world_joint" type="fixed">
    <parent link="world" />
    <child link = "cart" />
    <origin xyz="0.0 0.0 1.0" rpy="0.0 0.0 0.0" />
  </joint>

  <link name = "cart"/>

  <joint name="cart_joint" type="fixed">
    <parent link="cart" />
    <child link = "base_link" />
    <origin xyz="0.0 0.7 0.0" rpy="1.570796 0.0 1.570796" /> 
  </joint>


  <xacro:ur5_robot prefix="" joint_limited="true"
    shoulder_pan_lower_limit="${-pi}" shoulder_pan_upper_limit="${pi}"
    shoulder_lift_lower_limit="${-pi}" shoulder_lift_upper_limit="${pi}"
    elbow_joint_lower_limit="${-pi}" elbow_joint_upper_limit="${pi}"
    wrist_1_lower_limit="${-pi}" wrist_1_upper_limit="${pi}"
    wrist_2_lower_limit="${-pi}" wrist_2_upper_limit="${pi}"
    wrist_3_lower_limit="${-pi}" wrist_3_upper_limit="${pi}"
  />
<!-- Attach Kinect -->  
  <xacro:sensor_kinect parent="ee_link" />
  <!-- <xacro:kinect2 parent="ee_link" name="KinectSensor2">
    <origin xyz="0 0 0" rpy="0 0 0" />
  </xacro:kinect2> -->

<!-- Gionni was here -->
</robot>

